canopy_bioemi_mod.F90

```diff
@@ -0,0 +1,310 @@
+        REAL(RK),          PARAMETER     :: DTEMP_5_SUN     =  -0.535_rk  !...
+        REAL(RK),          PARAMETER     :: ATEMP_1_SHADE   =  -12.846_rk !...
+        REAL(RK),          PARAMETER     :: ATEMP_2_SHADE   =  -11.343_rk !...
+        REAL(RK),          PARAMETER     :: ATEMP_3_SHADE   =  -1.068_rk  !...
+        REAL(RK),          PARAMETER     :: ATEMP_4_SHADE   =  -5.551_rk  !...
+        REAL(RK),          PARAMETER     :: ATEMP_5_SHADE   =  -5.955_rk  !...
+        REAL(RK),          PARAMETER     :: BTEMP_1_SHADE   =  1.060_rk   !...
+        REAL(RK),          PARAMETER     :: BTEMP_2_SHADE   =  1.053_rk   !...
+        REAL(RK),          PARAMETER     :: BTEMP_3_SHADE   =  1.031_rk   !...
+        REAL(RK),          PARAMETER     :: BTEMP_4_SHADE   =  1.051_rk   !...
+        REAL(RK),          PARAMETER     :: BTEMP_5_SHADE   =  1.053_rk   !...
+        REAL(RK),          PARAMETER     :: CTEMP_1_SHADE   =  0.871_rk   !...
+        REAL(RK),          PARAMETER     :: CTEMP_2_SHADE   =  0.890_rk   !...
+        REAL(RK),          PARAMETER     :: CTEMP_3_SHADE   =  0.916_rk   !...
+        REAL(RK),          PARAMETER     :: CTEMP_4_SHADE   =  0.941_rk   !...
+        REAL(RK),          PARAMETER     :: CTEMP_5_SHADE   =  0.956_rk   !...
+        REAL(RK),          PARAMETER     :: DTEMP_1_SHADE   =  0.015_rk   !...
+        REAL(RK),          PARAMETER     :: DTEMP_2_SHADE   =  -0.141_rk  !...
+        REAL(RK),          PARAMETER     :: DTEMP_3_SHADE   =  -0.368_rk  !...
+        REAL(RK),          PARAMETER     :: DTEMP_4_SHADE   =  -0.592_rk  !...
+        REAL(RK),          PARAMETER     :: DTEMP_5_SHADE   =  -0.743_rk  !...
+
+        REAL(RK),          PARAMETER     :: CT2             =  230.0_rk   !Deactivation energy (kJ/mol) (Guenther et al., 2012)
+
+!Calculate photolyis shading/correction factor through canopy, i.e., the fraction of sunlit leaves downward through canopy
+
+        call canopy_phot(FCLAI, LAI, CLU, COSZEN, FSUN)
+
+! Use linear canopy temperature model based on Silva et al. (2020) to get approx. sun/shade leaf temperatures
+! through canopy (ignores effect of wind speed on leaf boundary layer ~ 1 % error/bias)
+!Citation:
+!Silva, S. J., Heald, C. L., and Guenther, A. B.: Development of a reduced-complexity plant canopy
+!physics surrogate model for use in chemical transport models: a case study with GEOS-Chem v12.3.0,
+!Geosci. Model Dev., 13, 2569–2585, https://doi.org/10.5194/gmd-13-2569-2020, 2020.
+        do i=1, SIZE(ZK)  !calculate linear change in parameters interpolated to Silva et al. 5 layer canopy regions
+            if (ZK(i) .gt. FCH) then ! above canopy, Tleaf = Tair
+                ATEMP_SUN(i)   = 0.0
+                BTEMP_SUN(i)   = 1.0
+                ATEMP_SHADE(i) = 0.0
+                BTEMP_SHADE(i) = 1.0
+            else if (ZK(i) .le. FCH .and. ZK(i) .gt. FCH*(4.0_rk/5.0_rk)) then  !Level 1 - 2
```

Comment:
where did you get these boundary definitions? I don't see that in the paper

Reply:
@zmoon They are not in the paper, and I use them because in the linear Tleaf and exponential PPFDleaf equations using these boundary conditions gives back the T2 and PPFD above canopy.  

Tleaf = Tair (when A = 1 and B = 0)
PPFD = PPFDtoc (when C=0 and D=0)

Reply:
Sure, I meant the z boundaries. The $4/5 h_c$ and so on.

Reply:
> Do you want to give it a shot?

Yeah probably later, for this PR let's focus on what you have already.

Reply:
@zmoon   Sounds good, but I agree we should bring in some aspect of canopy layer and ground energy balance models into Canopy-App following other applications of MEGAN, Bonan's many works, or other applications of canopy parameterizations in gridded models, like [WRF MCANOPY](https://agupubs.onlinelibrary.wiley.com/doi/pdf/10.1029/2018MS001347).  This will allow us to do more sophisticated parameterizations of biogenics, deposition, soil fluxes, and things like biogeochemical fluxes of carbon etc.

Reply:
We will come back to this for sure.